### PR TITLE
docs: update linking to aws-lc-rs crate etc.

### PR DESCRIPTION
**Pull request by brodycj**
- add add mono-spaced linking to the `aws-lc-rs` crate in several places in `README.md` & `rustls/src/lib.rs`
- add small clarification of when `aws-lc-rs` provider would be available

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ list of protocol features](https://docs.rs/rustls/latest/rustls/manual/_04_featu
 
 While Rustls itself is platform independent, by default it uses [`aws-lc-rs`] for implementing
 the cryptography in TLS.  See [the aws-lc-rs FAQ][aws-lc-rs-platforms-faq] for more details of the
-platform/architecture support constraints in aws-lc-rs.
+platform/architecture support constraints in [`aws-lc-rs`].
```
Comment (djc) [inline README.md:48]: I don't think we commonly spell crate names as literals (at least I personally normally don't use code literals for crate names, except when used in the context of actual code).

**Review by djc — COMMENTED** [review]
This doesn't seem like a material improvement to me.

**Comment by brodycj** [comment]
Yes this is really minor, I thought this could be a small improvement in terms of consistency. I also thought that anything that could mean something to a computer (normal computer, not something that is using AI), like a function name or crate name, would often be shown with monospacing. I rest my case, don't mind if we just close this.

File: README.md (hunk 1 above)
Comment (ctz) [reply README.md:48]: Yeah, agree. Probably some instants of not following this rule elsewhere in the crate.

**Pull request closed without merging**